analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -582,6 +663,256 @@ void AsOperatorWithNonNullLiteral(int? i)
+
+        nullableField = null;
+        ModifyOutParam(out nullableField);
+        _ = nullableField.Value;                         // Compliant
+
+        ModifyOutParam(out var newLocal);
+        _ = newLocal.Value;                             // Unknown
+        newLocal = null;
+        _ = newLocal.Value;                             // Noncompliant
+
+        static void ModifyOutParam(out int? i) => i = null;
+        static void ModifyOutParamAndRead(out int? i1, int? i2) => i1 = i2;
+        static void ReadAndModifyOutParam(int? i1, out int? i2) => i2 = i1;
+    }
+
+    void RefParams(int? iParam)
+    {
+        iParam = null;
+        ModifyRefParam(ref iParam);
+        _ = iParam.Value;                                // Compliant, unknown after method call
+
+        int? iLocal = null;
+        ModifyRefParam(ref iLocal);
+        _ = iLocal.Value;                                // Compliant
+
+        iLocal = null;
+        ModifyRefParamAndRead(ref iLocal, iLocal.Value); // FN
+
+        iLocal = null;
+        ReadAndModifyRefParam(iLocal.Value, ref iLocal); // Noncompliant
+
+        nullableField = null;
+        ModifyRefParam(ref nullableField);
+        _ = nullableField.Value;                         // Compliant
+
+        static void ModifyRefParam(ref int? i) => i = null;
+        static void ModifyRefParamAndRead(ref int? i1, int? i2) => i1 = i2;
+        static void ReadAndModifyRefParam(int? i1, ref int? i2) => i2 = i1;
+    }
+}
+
+class MutableField
+{
+    int? theField;
+
+    void Basics()
+    {
+        _ = theField.Value;          // Compliant, unknown
+
+        theField = null;
+        _ = theField.Value;          // Noncompliant, empty
+        _ = theField.Value;          // Compliant, when reached the ".Value" above implies theField is not null
+    }
+
+    void LocalFunctions()
+    {
+        theField = 42;
+        LocalFunctionChangingTheField();
+        _ = theField.Value;          // Compliant, unknown
+
+        theField = null;
+        LocalFunctionChangingTheField();
+        _ = theField.Value;          // Noncompliant, FP, local functions don't reset the state
+
+        theField = 42;
+        LocalFunctionNotChangingTheField();
+        _ = theField.Value;          // Compliant, unknown
+
+        void LocalFunctionChangingTheField() => theField = null;
+        void LocalFunctionNotChangingTheField() { }
+    }
+
+    void Methods()
+    {
+        theField = 42;
+        MethodChangingTheField();
+        _ = theField.Value;          // Compliant, unknown
+
+        theField = null;
+        MethodChangingTheField();
+        _ = theField.Value;          // Compliant, instance methods reset the state
+
+        theField = 42;
+        MethodNotChangingTheField();
+        _ = theField.Value;          // Compliant, unknown
+
+        void LocalFunctionChangingTheField() => theField = null;
+        void LocalFunctionNotChangingTheField() { }
```

Comment:
```suggestion
```

Reply:
Removed these two unused local functions in https://github.com/SonarSource/sonar-dotnet/pull/7006 (new PR for all issues detected on this PR but on code already merged into `feature/SE`).